### ISSUE TITLE
UNREACH doesn't seem to return correct value

### ISSUE THREAD

**Issue by infernix** [danielperna84/pyhomematic#121]
I'm trying to add tests to my HASS setup for when devices are offline (e.g. battery is dead, bad reception etc), but I'm not getting valid UNREACH data back:

```
root@homegearpi:~# homegear -r
Connected to Homegear (version 0.7.13-1549).
> fs 0
Device family "HomeMatic BidCoS" selected.
For information about the family's commands type: "help"
(Family)> ls
         ID │ Name                      │  Address │ Serial Number │ Type │ Type String               │ Firmware │ Config Pending │ Unreach │ Low Bat
────────────┼───────────────────────────┼──────────┼───────────────┼──────┼───────────────────────────┼──────────┼────────────────┼─────────┼────────
            │                           │          │               │      │                           │          │                │         │        
          1 │ Thermostat Living         │   47402E │    NEQ0362154 │ 00AD │ HM-TC-IT-WM-W-EU          │      1.3 │             No │      No │      No
...
         21 │ Thermostat Bathroom       │   3679BD │    MEQ0235295 │ 00AD │ HM-TC-IT-WM-W-EU          │      1.3 │             No │     Yes │     Yes
────────────┴───────────────────────────┴──────────┴───────────────┴──────┴───────────────────────────┴──────────┴────────────────┴─────────┴────────
(Family)> 
```

MEQ0235295 is clearly shown as UNREACH as well as LOWBAT. But:

```
Python 3.5.3 (default, Jan 19 2017, 14:11:04) 
[GCC 6.3.0 20170118] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyhomematic
>>> def syscb(src, *args):
...     print(src)
...     for arg in args:
...         print(arg)
... 
>>> def cb1(address, interface_id, key, value):
...     print("CALLBACK WITH CHANNELS: %s, %s, %s, %s" % (address, interface_id, key, value))
... 
>>> def cb2(address, interface_id, key, value):
...     print("CALLBACK WITHOUT CHANNELS: %s, %s, %s, %s" % (address, interface_id, key, value))
... 
>>> from pyhomematic import HMConnection
>>> pyhomematic = HMConnection(local="192.168.0.1",localport=7001,remote="192.168.0.11",remoteport=2001, systemcallback=syscb) # Create server thread
createDeviceObjects
>>> pyhomematic.start()
True
...
>>> 
>>> pyhomematic.devices['default']['NEQ0362154'].UNREACH
False
>>> pyhomematic.devices['default']['MEQ0235295'].UNREACH
False
```

Part of the problem is that the battery state data is useless:
```
>>> pyhomematic.devices['default']['NEQ0362154'].battery_state()
2.8
>>> pyhomematic.devices['default']['MEQ0235295'].battery_state()
2.1
```

And lowbat, despite being shown in homegear, seems not supported for thermostats:
```
>>> pyhomematic.devices['default']['NEQ0362154'].low_batt()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'ThermostatWall' object has no attribute 'low_batt'
```
So there's no way to really find out with hass/pyhomematic if a device battery is dead, nor if it is reachable.

Is there something I'm missing here?

**Comment by danielperna84** [comment]
To automated based on the battery you hae to use templates which look at the attributes of the entity. An example of that can be found in the documentation here: https://home-assistant.io/components/homematic/#reading-attributes-of-entities

The whole purpose is to not do this in Python. Different devices expose the battery state in different ways. So the usual route would always be to look at the attributes of your devices and use them.

The exception to this would be the unreachable-state. I have to look into that.

**Comment by infernix** [comment]
I fully understand the use of template sensors and automation based on entity attributes.

The point i am making is that there is no attribute that is currently exposed which allows me to alert on a thermostat that is unreachable or reporting low battery. The _only_ attribute available is battery_state which current reports 2.1 for a thermostat which is not working due to the batteries being dead. So that's quite useless unfortunately. 

So there is currently no way to run any automation or to create a binary template sensor as an indicator for this. Having UNREACH correctly exposed in homematic component for HASS is perfect, but I can't even get that to work within pyhomematic. But it's correctly displayed in homegear, which is what I don't get :)

**Comment by danielperna84** [comment]
Different devices report the battery state differently. Shutter-contacts for example often have `LOWBAT` set to `True` if the battery is low. Thermostats however don't do that. Instead they report the remaining power. If your Thermostats battery is dead at a value of 2.1, then your automation has to monitor that value and fire an automation as soon as the value drops below 2.2.

The UNREACH is a totally different topic. This may not be properly implemented because it had a low priority. Ideally devices are never unreachable, so this hasn't received much attention.
